### PR TITLE
seed project

### PR DESCRIPTION
**Pull request by MeerKatDev**

### CODE REVIEW

File: src/main/scala/Application.scala [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    if(!errorMessage.isEmpty) {
+      baseObject
+      .put("status", "error")
+      .put("message", errorMessage)
+    } else {
+      val testSuite = getTestSuiteObject(filepath)
+      val failuresNum = testSuite.getInt("failures")
+      val testCasesArray = testSuite.getJSONArray("testcase")
+      // val fileSource = Source.fromURL(logFilePath)
+      Console.err.println("asdasd")
```
Comment (ErikSchierboom) [inline src/main/scala/Application.scala:38]: I think this should be removed

File: src/main/scala/Application.scala [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+        .put("name", o.getString("name"))
+        .put("status", if(fail != null) "fail" else "pass" )
+        .put("message", if(fail != null) fail.getString("message") else "")
+        .put("output", "TOIMPLEMENT")
+        .put("test_code", "TOIMPLEMENT")
+      })
+
+      baseObject
+      .put("status", if(failuresNum > 0) "fail" else "pass")
+      .put("message", "")
```
Comment (ErikSchierboom) [inline src/main/scala/Application.scala:52]: As per [the docs](https://github.com/exercism/docs/blob/main/anatomy/track-tooling/test-runners/interface.md#message):

> When the status is not `error`, either set the value to `null` or omit the key entirely.

I'd suggest to just omit the key entirely:

```suggestion
```

File: src/test/scala/ApplicationSpec.scala [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+  }
+
+  "A successful xml" should "be properly formatted as JSON" in {
+    val xmlTestURL = getClass.getResource("/GradeSchool_successful.xml")
+    val outputFileURL = getClass.getResource("/outputs/output.txt")
+    val exercismOutput: JSONObject = Application.toExercismJSON(xmlTestURL, outputFileURL)
+    assert(exercismOutput.getInt("version") == 2)
+    assert(exercismOutput.getString("status") == "pass")
+    assert(exercismOutput.getString("message") == "")
+    assert(exercismOutput.get("tests").asInstanceOf[Array[JSONObject]].length == 7)
```
Comment (ErikSchierboom) [inline src/test/scala/ApplicationSpec.scala:23]: I would suggest also verifying the exact contents of the `tests` field, to verify that the keys of each element in the `tests` array has the proper value (most importantly the `name` and the `status`). This comment also applies to the other tests.

File: src/main/scala/Application.scala [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+    val errorMessage = findErrorsInLog(logFilePath)
+    if(!errorMessage.isEmpty) {
+      baseObject
+      .put("status", "error")
+      .put("message", errorMessage)
+    } else {
+      val testSuite = getTestSuiteObject(filepath)
+      val failuresNum = testSuite.getInt("failures")
+      val testCasesArray = testSuite.getJSONArray("testcase")
+      // val fileSource = Source.fromURL(logFilePath)
```
Comment (ErikSchierboom) [inline src/main/scala/Application.scala:37]: Maybe remove this comment?

**Review by ErikSchierboom — APPROVED** [review]
Excellent work! I've left some comments.

File: src/main/scala/Application.scala (hunk 2 above)
Comment (MeerKatDev) [reply src/main/scala/Application.scala:52]: I'd rather keep it - in FP it's best practice to always return some value, and moreover it's always the best, IMHO, to keep all the parameters, even if null. This way you avoid errors such as "the key is inexistent".

Comment (ErikSchierboom) [reply src/main/scala/Application.scala:52]: > in FP it's best practice to always return some value

I don't see how removing this field affects that, as you're still returning a value 🤷 

> it's always the best, IMHO, to keep all the parameters, even if null.

I'm fine with that, as long as it null and not an empty string.

Comment (MeerKatDev) [reply src/main/scala/Application.scala:52]: sure, I will make it a `null` then :)

File: src/test/scala/ApplicationSpec.scala (hunk 3 above)
Comment (MeerKatDev) [reply src/test/scala/ApplicationSpec.scala]: I added a new commit, is that what you meant? I thought I could assert field by field, when I add also `output` and `test_case`, in a second PR.

File: src/test/scala/ApplicationSpec.scala [hunk 5]
```diff
@@ -17,17 +17,27 @@ class ApplicationSpec extends AnyFlatSpec with Matchers {
...
+
     assert(exercismOutput.getInt("version") == 2)
     assert(exercismOutput.getString("status") == "pass")
-    assert(exercismOutput.getString("message") == "")
-    assert(exercismOutput.get("tests").asInstanceOf[Array[JSONObject]].length == 7)
+    assert(exercismOutput.opt("message") == None)
+
+    val testCases = exercismOutput.get("tests").asInstanceOf[Array[JSONObject]]
+    assert(testCases.length == 7)
+    assert(testCases(0).toString() == """{"output":"TOIMPLEMENT","name":"empty school","test_code":"TOIMPLEMENT","message":"None","status":"pass"}""")
```
Comment (ErikSchierboom) [inline src/test/scala/ApplicationSpec.scala:27]: It looks like `"message":"None"` is not what you want, or is it?

File: src/test/scala/ApplicationSpec.scala (hunk 3 above)
Comment (ErikSchierboom) [reply src/test/scala/ApplicationSpec.scala]: Yeah sure. I like how the new test verifies the contents.

File: src/test/scala/ApplicationSpec.scala (hunk 5 above)
Comment (MeerKatDev) [reply src/test/scala/ApplicationSpec.scala]: Indeed it wasn't :)

**Pull request merged**